drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/nodes/vector/VectorValueNode.java

```diff
@@ -160,30 +159,53 @@ private void generateVectorAssign(NodeLIRBuilderTool gen, LIRGeneratorTool tool,
 
         Value s0,s1,s2,s3,s4,s5,s6,s7;
         switch (kind.getVectorLength()) {
-            case 2:
+            case 2: {
                 final OCLOp2 op2 = VectorUtil.resolveAssignOp2(getOCLKind());
                 s0 = getParam(gen, tool, 0);
+                OCLKind kindOp = (OCLKind) s0.getPlatformKind();
+                if (kindOp.isVector()) {
+                    gen.setResult(this, s0);
+                    return;
+                }
```

Comment:
Can we move this in a method since it is used in every case?
I am not sure about the naming, but it could be something like `setResultIfVector(NodeLIRBuilder gen, Value s0)`.